x-pack/plugins/index_management/server/lib/fetch_indices.ts

```diff
@@ -76,8 +76,12 @@ async function fetchIndicesCall(
     query: catQuery,
   });
 
-  // The two responses should be equal in the number of indices returned
-  return catHits.map((hit) => {
+  // System indices may show up in _cat APIs, as these APIs are primarily used for troubleshooting
+  // For now, we filter them out and only return index information for the indices we have
+  // In the future, we should migrate away from using cat APIs (https://github.com/elastic/kibana/issues/57286)
+  const filteredCatHits = catHits.filter((hit) => typeof indices[hit.index] !== 'undefined');
```

Comment:
This might be a bit silly, but I'm wondering if performance could be a concern. We know users might have hundreds of thousands (maybe millions) of indices. In this case we are doubling memory consumption by creating the `filteredCatHits` variable, and also doubling the number of iterations by adding another `filter` loop. There's a good chance that testing could reveal this to be a non-issue. But we could also just make the code immune to these concerns by sticking with a single loop and doing our `undefined` check within the loop:

```js
return catHits.reduce((decoratedIndices, hit) => {
  const index = indices[hit.index];
  
  if (typeof index !== 'undefined') {
    const aliases = Object.keys(index.aliases);

    decoratedIndices.push({
      health: hit.health,
      status: hit.status,
      name: hit.index,
      uuid: hit.uuid,
      primary: hit.pri,
      replica: hit.rep,
      documents: hit['docs.count'],
      size: hit['store.size'],
      isFrozen: hit.sth === 'true', // sth value coming back as a string from ES
      aliases: aliases.length ? aliases : 'none',
      hidden: index.settings.index.hidden === 'true',
      data_stream: index.data_stream,
    });
  }
  
  return decoratedIndices;
}, []);
```

Reply:
👍 great point. Thanks for fixing!